ocs_ci/ocs/scale_pgsql.py

```diff
@@ -0,0 +1,141 @@
+    """
+    Scale Postgresql workload with scale parameters and functions
+    """
+
+    def __init__(self, node_selector=constants.SCALE_NODE_SELECTOR, **kwargs):
+        """
+        Initializer function
+
+        """
+        super().__init__(**kwargs)
+        self._node_selector = node_selector
+
+    def apply_crd(self, crd):
+        Postgresql.apply_crd(self, crd=crd)
+
+    def setup_postgresql(self, replicas, node_selector=None):
+        # Node selector for postgresql
+        pgsql_sset = templating.load_yaml(
+            constants.PGSQL_STATEFULSET_YAML
+        )
+        if node_selector is not None:
+            pgsql_sset['spec']['template']['spec'][
+                'nodeSelector'] = node_selector
+        Postgresql.setup_postgresql(self, replicas=replicas)
+
+    def _create_pgbench_benchmark(
+        self, replicas, clients=None, threads=None,
+        transactions=None, scaling_factor=None,
+        timeout=None
+    ):
+        Postgresql.create_pgbench_benchmark(
+            self, replicas, clients=clients,
+            threads=threads, transactions=transactions,
+            scaling_factor=scaling_factor, timeout=timeout
+        )
+
+    def _wait_for_pgbench_status(self, status, timeout=None):
+        Postgresql.wait_for_postgres_status(
+            self, status=status, timeout=timeout
+        )
+
+    def _get_pgbench_pods(self):
+        Postgresql.get_pgbench_pods()
+
+    def _validate_pgbench_run(self, pgbench_pods, print_table=True):
+        Postgresql.validate_pgbench_run(self, pgbench_pods, print_table=True)
+
+    def cleanup(self):
+        # Cleanup postgresql
+        Postgresql.cleanup(self)
+
+        # Remove scale label and delete machineset
+        delete_worker_node()
+
+
+def add_worker_node(instance_type=None):
+    # Add worker node to cluster
+    log.info("Adding worker nodes on the current cluster")
+    global ms_name
+    ms_name = list()
+    dt = config.ENV_DATA['deployment_type']
+    if dt == 'ipi':
+        # Create machineset for app worker nodes on each zone
+        for obj in machine.get_machineset_objs():
+            if 'app' in obj.name:
+                ms_name.append(obj.name)
+        if instance_type is not None:
+            instance_type = instance_type
+        else:
+            instance_type = 'm5.4xlarge'
+        if not ms_name:
+            if len(machine.get_machineset_objs()) == 3:
+                for zone in ['a', 'b', 'c']:
+                    ms_name.append(
+                        machine.create_custom_machineset(
+                            instance_type=instance_type, zone=zone
+                        )
+                    )
+            else:
+                ms_name.append(
+                    machine.create_custom_machineset(
+                        instance_type=instance_type, zone='a'
+                    )
+                )
+            for ms in ms_name:
+                machine.wait_for_new_node_to_be_ready(ms)
```

Comment:
@tiffanyn108 I think you can replace this with below code, which will check if there is enough worker node in setup if there is enough workers then it will continue the TC without adding more worker, if not it will add the worker based on the expected worker count.
```
        from ocs_ci.ocs import scale_lib
        # Check for expected worker count
        expected_worker_count = scale_lib.get_expected_worker_count(scale_count)
        if scale_lib.check_and_add_enough_worker(expected_worker_count):
            if config.ENV_DATA['deployment_type'] == 'ipi' and config.ENV_DATA['platform'].lower() == 'aws':
                for obj in machine.get_machineset_objs():
                    if 'app' in obj.name:
                        ms_name.append(obj.name)
            else:
                ms_name = []
```

Reply:
This will be addressed in future PR which will be a common function to use for all the workloads.